File: 1055-pairs-of-songs-with-total-durations-divisible-by-60/pairs-of-songs-with-total-durations-divisible-by-60.py
class Solution:
    def numPairsDivisibleBy60(self, time: list[int]) -> int:
        rems = {}
        count = 0
        for t in time:
            rem = (t % 60) % 60
            target = (60 - rem) % 60
            if target in rems:
                count += rems[target]
            rems[rem] = rems.get(rem, 0) + 1
        return count

File: 1055-pairs-of-songs-with-total-durations-divisible-by-60/pairs-of-songs-with-total-durations-divisible-by-60.c
int numPairsDivisibleBy60(int* time, int timeSize) {
    int rems[60] = {0};
    int count = 0;
    for (int i = 0; i < timeSize; ++i) {
        int cur_rem = time[i] % 60;
        int target_rem = (60 - cur_rem) % 60;
        count += rems[target_rem];
        rems[cur_rem] += 1;
    }
    return count;
}